_inc/lib/class-jetpack-tweetstorm-helper.php

```diff
@@ -1413,8 +1419,16 @@ private static function extract_tag_content_from_html( $tags, $html ) {
 						$offset += self::$characters_per_url - strlen( $url );
 
 						// If we're in a link with a URL set, there's no need to keep two copies of the same link.
-						if ( $url === $current_url ) {
-							$current_url = '';
+						if ( ! empty( $current_url ) ) {
+							if ( $url === $current_url ) {
+								$current_url = '';
+							}
+
+							// Check that the link text isn't just a shortened version of the href value.
+							$trimmed_current_url = preg_replace( '|^https?://|', '', $current_url );
+							if ( $url === $trimmed_current_url || trim( $trimmed_current_url, '/' ) === $url ) {
```

Comment:
Maybe we should make this a case-insensitive comparison?

Reply:
Fixed in 74d4a27c5.